packages/typespec-ts/src/index.ts

```diff
@@ -403,8 +403,7 @@ export async function $onEmit(context: EmitContext) {
     );
     const hasPackageFile = await existsSync(existingPackageFilePath);
     const shouldGenerateMetadata =
-      option.generateMetadata === true ||
-      (option.generateMetadata === undefined && !hasPackageFile);
+      option.generateMetadata === true || !hasPackageFile;
```

Comment:
What if data-plane SDKs don't want to generate metadata so they turn off the option "generate-medadata=false"?

This may happen in local experience where service team needs to refresh their SDKs.